taipy/core/reason/reason.py

```diff
@@ -9,33 +9,114 @@
 # an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the License for the
 # specific language governing permissions and limitations under the License.
 
-from typing import Dict, Set
+from typing import Any, Optional
 
 
-class Reasons:
-    def __init__(self, entity_id: str) -> None:
-        self.entity_id: str = entity_id
-        self._reasons: Dict[str, Set[str]] = {}
+class Reason:
+    """
+    A reason holds the reason why specific actions cannot be performed.
```

Comment:
```suggestion
    A reason explains why a specific action cannot be performed.
    
    This is a parent class aiming at being implemented by specific sub-classes.
    
    Because Taipy applications are natively multiuser, asynchronous, and dynamic, 
    some functions might not be called in some specific contexts. You can protect 
    such calls by calling other methods that return `Reasons^` objects. A `Reasons^` 
    object acts like a boolean: True if the action can be performed and False otherwise. 
    If the action cannot be performed, the `Reasons^` object holds all the reasons why 
    the action cannot be performed as a list of `Reason` objects.
```